pysteps/motion/correlation.py

```diff
@@ -0,0 +1,330 @@
+# -*- coding: utf-8 -*-
+"""
+pysteps.motion.correlation
+========================
+
+Implementation of the classical correlation based motion vector.
+
+.. autosummary::
+    :toctree: ../generated/
+
+    correlation
+"""
+
+import numpy as np
+import math
+
+from pysteps.decorators import check_input_frames
+from pysteps.utils.cleansing import detect_outliers
+from pysteps import utils
+
+# To delete
+import pdb
+
+# Check if numba can be imported
+try:
+    from numba import jit
+
+    NUMBA_IMPORTED = True
+except ImportError:
+    NUMBA_IMPORTED = False
+
+
+@check_input_frames(2)
+def correlation(
+    input_images,
+    settings={},
+    interp_method="idwinterp2d",
+    interp_kwargs=None,
+    dense=True,
+    nr_std_outlier=3,
+    k_outlier=30,
+    verbose=False,
+):
+    """
+    Run the correlation based optical flow method and interpolate the motion
+    vectors. This is using NUMBA to speed up the correlation computation.
+
+    The sparse motion vectors are finally interpolated to return the whole
+    motion field.
+
+    Parameters
+    ----------
+    input_images: ndarray_
+        Array of shape (T, m, n) containing a sequence of *T* two-dimensional
+        input images of shape (m, n). The indexing order in **input_images** is
+        assumed to be (time, latitude, longitude).
+
+        *T* = 2 is the minimum required number of images.
+        With *T* > 2, all the resulting obtained vectors are pooled together for
+        the final interpolation on a regular grid.
+
+    settings: dict, optional
+        Optional dictionary containing keyword arguments for the correlation
+        based algorithm.
+
+    interp_method: {"idwinterp2d", "rbfinterp2d"}, optional

```

Comment:
do you think it could be added as an additional interpolator in this module? pysteps/utils/interpolate.py (I wouldn't mind the numba optional requirement)